src/Glpi/Log/LegacyGlobalLogger.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Glpi\Log;
+
+use Psr\Log\LoggerInterface;
+
+class LegacyGlobalLogger implements LoggerInterface
+{
+    private ?LoggerInterface $internalLogger = null;
+
+    private function getLogger(): LoggerInterface
+    {
+        if ($this->internalLogger) {
+            return $this->internalLogger;
+        }
+
+        global $PHPLOGGER;
+
+        if (!$PHPLOGGER) {
+            (new \GLPI())->initLogger();
+        }
+        if (!$PHPLOGGER instanceof LoggerInterface) {
+            throw new \RuntimeException('There was an unknown error preventing initialization of the legacy GLPI Logger.');
+        }
+
+        $this->internalLogger = $PHPLOGGER;
+
+        return $this->internalLogger;
+    }
```

Comment:
Not sure we should call the `GLPI::initLogger()` here. If it is call before configuration constants are set, it may result in an unexpected default log level.

Maybe it should return `$PHPLOGGER ?? new \Psr\Log\NullLogger()`.

```suggestion
    private function getLogger(): LoggerInterface
    {
        /**
         * @var \Psr\Log\LoggerInterface $PHPLOGGER
         */
        global $PHPLOGGER;
        return $PHPLOGGER ?? new \Psr\Log\NullLogger();
    }
```

Reply:
Good point 👌